app/models/vm/operations.rb

```diff
@@ -14,7 +14,11 @@ module Vm::Operations
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddresses.first, :port => 9090).to_s
+    if MiqServer.active_miq_servers.where(:has_active_cockpit_ws => true)
```

Comment:
I would much rather drive this with roles now that I am seeing how this all flows together.  That way a system with the appropriate role can do the work, and enabling it by the customer is consistent with enabling other services.